ql/src/test/queries/clientpositive/ptf_register_use_2.q

```diff
@@ -0,0 +1,2 @@
+-- session 2: use function
```

Comment:
Looks like your qtest depends on the two different session. I am not sure if it is a good way to create two qtest to achieve this, then this needs the two qtests must be ordered, that is to say `ptf_register_use_1.q` must be in front of `ptf_register_use_2.q`. But i think the qtests framework may not work as you expected sometimes, and will lead your test flaky.

I may suggest you adding the create function in https://github.com/apache/hive/blob/master/data/files/datasets/temp_udf/load.hive.sql (Like [HIVE-23558](https://github.com/apache/hive/pull/4928/files#top)), and add `--! qt:dataset:temp_udf` in another qtest to invoke this funtion.  This can aslo create two seesions and can reproduce your issue.

cc @ayushtkn may give some thought.

Reply:
Thanks for your review! As you suggest, this pr should add `data/files/datasets/temp_udaf/load.hive.sql` file to create function, and a `ptf_register_use.q` file to invoke function.  

I test it local. By log, hive does use two session to exeuct create an invoke.

```txt
2024-01-15T22:45:28,041  INFO [main] SessionState: Hive Session ID = bbc2a0ea-227a-4c07-a7a6-52c3a50e1e06
...
2024-01-15T22:45:29,451  INFO [main] SessionState: Hive Session ID = e8ffa509-9272-483b-a885-ef7d92e705cd
```

Also, the create does execute before invoke :
https://github.com/apache/hive/blob/06ef7c8231501c0314875d640f081adc35a70db6/itests/util/src/main/java/org/apache/hadoop/hive/cli/control/CoreCliDriver.java#L112-L115

So, are we agreed on the idea?

Reply:
Yes, if you don't have other better ways, let's use this idea. I think this idea is better than creating two qtests at least.

Reply:
Another question, can i use `git push --force` to amend this modify on hive repository?

Reply:
sure